src/config.rs

```diff
@@ -393,23 +393,103 @@ impl Config {
+
+            let before_caption = section_line.split_once(':').map(|s| s.0);
+            if let Some(section) = before_caption {
+                let mut parts = section.splitn(2, ' ');
+
+                // in case of component definition, section-kind will be mandatory
+                // -- <optional: section-kind> section-name: <section-caption>
+                return match (parts.next(), parts.next()) {
+                    (Some(_kind), Some(_name)) => true,
+                    (Some(_name), None) if _name.contains("youtube") => true,
+                    (_, _) => false,
+                };
+            }
+
+            false
+        }
+
+        /// returns true if the line is commented or escaped out section, otherwise false
+        fn is_section_commented_or_escaped(line: &str) -> bool {
+            line.starts_with("/-- ") || line.starts_with(r"\-- ")
+        }
+
+        /// returns true if the line is commented out section/subsection, otherwise false
+        fn is_line_commented(line: &str) -> bool {
+            line.starts_with("/-- ") || line.starts_with("/--- ")
+        }
+
+        /// returns true if the line is escaped out section/subsection, otherwise false
+        fn is_line_escaped(line: &str) -> bool {
+            line.starts_with(r"\-- ") || line.starts_with(r"\--- ")
+        }
+
+        // id: <some-id>
+        // <some-id> could be any string using this character set
+        // character set = [A-Z, a-z, 0-9, whitespace, '_' , '-' ]
+        // leading and trailing whitespace characters are ignored
+        const ID_HEADER_PATTERN: &str = r"(?m)^\s*id\s*:[-_\sA-Za-z\d]*$";
         lazy_static::lazy_static!(
             static ref ID: regex::Regex = regex::Regex::new(ID_HEADER_PATTERN).unwrap();
         );
 
-        // grep all lines where user defined `id` for the sections
+        // Flags to ignore grepping for id under certain cases
+        let mut ignore_id: bool = true;
+        let mut register_id_for_last_section: bool = false;
+
+        // grep all lines where user defined `id` for the sections/ subsecions
         // and update the global_ids map
         for (ln, line) in itertools::enumerate(data.lines()) {
-            if ID.is_match(line) {
+            // Trim leading whitespaces if any
+            let line = line.trim_start();
+
+            // Ignore for commented out section/subsection
+            // and for ftd code passed down as body to ft.code
+            if is_line_commented(line) || is_line_escaped(line) {
+                if is_section_commented_or_escaped(line) {
+                    register_id_for_last_section = false;
+                }
+                ignore_id = true;
+            }
+
+            // section could be component definition
+            // in that case ignore relevant id's defined under its definition
+            // including the ids defined on its subsections
+            if line.starts_with("-- ") {
```

Comment:
`--` this also come from `ftd`, like `SECTION_IDENTIFIER`. Or even you should ask it from ftd, is `ftd::section::start_with(line, ftd::SECTION_IDENTIFIER)`.